repo2docker/buildpacks/r.py

```diff
@@ -0,0 +1,156 @@
+import re
+import os
+import datetime
+
+from .python import PythonBuildPack
+
+class RBuildPack(PythonBuildPack):
+    """
+    Setup R for use with a repository
+
+    This sets up R + RStudio + IRKernel for a repository that contains:
+
+     1. A `runtime.txt` file with the text:
+
+        r-<year>-<month>-<date>
+
+        Where 'year', 'month' and 'date' refer to a specific
+        date snapshot of https://mran.microsoft.com/timemachine
+        from which libraries are to be installed.
+
+     2. An optional `install.R` file that will be executed at build time,
+        and can be used for installing packages from both MRAN and GitHub.
+
+    It currently sets up R from the ubuntu repository being used. This
+    is unideal, and we should investigate other solutions!
+    """
+    @property
+    def runtime(self):
+        """
+        Return contents of runtime.txt if it exists, '' otherwise
+        """
+        if not hasattr(self, '_runtime'):
+            runtime_path = self.binder_path('runtime.txt')
+            try:
+                with open(runtime_path) as f:
+                    self._runtime = f.read().strip()
+            except FileNotFoundError:
+                self._runtime = ''
+
+        return self._runtime
+
+    @property
+    def checkpoint_date(self):
+        """
+        Return the date of MRAN checkpoint to use for this repo
+
+        Returns '' if no date is specified
+        """
+        if not hasattr(self, '_checkpoint_date'):
+            match = re.match(r'r-(\d\d\d\d)-(\d\d)-(\d\d)', self.runtime)
+            if not match:
+                self._checkpoint_date = False
+            else:
+                self._checkpoint_date = datetime.date(*[int(s) for s in match.groups()])
+
+        return self._checkpoint_date
+
+    def detect(self):
+        """
+        Check if current repo should be built with the R Build pack
+
+        Note that we explicitly do *not* check if a requirements.txt
```

Comment:
I'm confused as I'm not sure where the super().detect() call is. I'm assuming it is called by the parent `PythonBuildPack`. Would be nice to clarify in the docstring.

If I am a user that uses conda, I'm assuming the `CondaBuildPack` can set up R.

Reply:
I'll try to clarify in another commit. The point I was trying to make is that we explcitly do *not* call super().detect() here, since that will fail unless there is a requirements.txt file. It's not very clear here, will redo!

And yes, conda has R support with environment.yml so no extra steps are needed. Mixing conda's R and other R probably causes problems.

Reply:
I got that we were not calling super().detect() explicitly. I didn't get why ;-). 

Perhaps something like:

Note: super().detect() is not called in this function. Calling it here
would fail unless a `requirements.txt` file exists, and that is not
a guaranteed case.

> Mixing conda's R and other R probably causes problems.

I would imagine that it would.